RevenueCatUI/Templates/Components/ScreenConditionObserver.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ScreenConditionObserver.swift
+//
+//  Created by Josh Holtz on 10/27/24.
+
+import Combine
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+enum ScreenCondition {
+    case `default`, mobileLandscape, tablet, tabletLandscape, desktop
+}
+
+class ScreenConditionObserver: ObservableObject {
+
+    private enum Device {
+        case phone, tablet, desktop
+    }
+
+    @Published var conditionType: ScreenCondition = .default
+
+    private let deviceType: Device
+    private var cancellable: AnyCancellable?
+
+    init() {
+        #if os(iOS)
+        // Set device type based on user interface idiom for iOS
+        switch UIDevice.current.userInterfaceIdiom {
+        case .phone:
+            deviceType = .phone
+        case .pad:
+            deviceType = .tablet
+        default:
+            deviceType = .phone
+        }
+        #else
+        // Set device type for macOS
+        deviceType = .desktop
+        #endif
```

Comment:
Does `userInterfaceIdiom` take into account the window size? I suppose iOS apps on macOS are resizable? Also apps don't need to run in full screen on iPads if I'm not mistaken. The device and orientation is not enough to determine how big our window is.